pkg/controller/constraint/constraint_controller.go

```diff
@@ -130,8 +129,7 @@ func (r *ReconcileConstraint) Reconcile(request reconcile.Request) (reconcile.Re
 		// Handle deletion
 		if containsString(finalizerName, instance.GetFinalizers()) {
 			if _, err := r.opa.RemoveConstraint(context.Background(), instance); err != nil {
-				// TODO make the unrecognized constraint error detectable via class introspection
-				if !(strings.HasPrefix(err.Error(), "Constraint kind ") && strings.HasSuffix(err.Error(), " is not recognized")) {
+				if reflect.TypeOf(err).String() == "client.UnrecognizedConstraintError" {
```

Comment:
Type switching would be the better approach (no strings):

if _, ok :=  err.(*client.UnrecognizedConstraintError); ok {

}

Reply:
Thanks @maxsmythe  - just changed it